server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -345,12 +367,38 @@ export class ScribeLambda implements IPartitionLambda {
             }
         }
 
-        const checkpoint = this.generateCheckpoint(message.offset);
-        this.checkpointCore(
-            checkpoint,
-            message,
-            this.clearCache);
-        this.lastOffset = message.offset;
+        // Create the checkpoint
+        this.updateCheckpointMessages(message);
+        this.checkpointInfo.rawMessagesSinceCheckpoint++;
+
+        if (this.noActiveClients) {
+            this.prepareCheckpoint(message, ScribeCheckpointReason.NoClients);
+            this.noActiveClients = false;
+        } else {
+            const checkpointReason = this.getCheckpointReason();
+            if (checkpointReason !== undefined) {
+                // checkpoint the current up-to-date state
+                this.prepareCheckpoint(message, checkpointReason);
+            } else {
+                this.updateCheckpointIdleTimer();
+            }
+        }
+    }
+
+    public prepareCheckpoint(message: IQueuedMessage, checkpointReason: ScribeCheckpointReason) {
+        // Get checkpoint context
+        const checkpoint = this.generateScribeCheckpoint(message.offset);
+        this.updateCheckpointMessages(message);
+
+        // write the checkpoint with the current up-to-date state
+            this.checkpoint(checkpointReason);
+            this.checkpointCore(
+                checkpoint,
+                message,
+                this.clearCache);
+            this.lastOffset = message.offset;
+            const checkpointResult = `Writing checkpoint. Reason: ${ScribeCheckpointReason[checkpointReason]}`;
+            Lumberjack.info(checkpointResult, getLumberBaseProperties(this.documentId, this.tenantId));
```

Comment:
Instead of adding the checkpoint reason within the message itself, do we instead to add it as a property, which could make this typo of info more "query-friendly"?

Reply:
I updated this so that the checkpoint reason is included as a property